packages/client/hmi-client/src/components/workflow/ops/model-config/tera-model-config-node.vue

```diff
@@ -12,59 +12,61 @@
 </template>
 
 <script setup lang="ts">
-import { cloneDeep, isEmpty } from 'lodash';
+import { cloneDeep } from 'lodash';
 import { computed, watch } from 'vue';
 import { WorkflowNode, WorkflowPortStatus } from '@/types/workflow';
 import Button from 'primevue/button';
 import TeraOperatorPlaceholder from '@/components/operator/tera-operator-placeholder.vue';
 import { getModel, getModelConfigurationsForModel } from '@/services/model';
 import { postAsConfiguredModel } from '@/services/model-configurations';
-import { ModelConfigOperationState } from './model-config-operation';
+import { ModelConfigOperation, ModelConfigOperationState, blankModelConfig } from './model-config-operation';
 
 const props = defineProps<{
 	node: WorkflowNode<ModelConfigOperationState>;
 }>();
-const emit = defineEmits(['open-drilldown', 'append-input-port', 'update-state']);
+const emit = defineEmits(['open-drilldown', 'append-input-port', 'update-state', 'append-output']);
 
 const modelInput = props.node.inputs.find((input) => input.type === 'modelId');
 const isModelInputConnected = computed(() => modelInput?.status === WorkflowPortStatus.CONNECTED);
 
 // Update the node with the new input ports
 watch(
 	() => props.node.inputs,
-	() => {
+	async () => {
+		const state = cloneDeep(props.node.state);
 		const inputs = props.node.inputs;
+
 		const documentInputs = inputs.filter((input) => input.type === 'documentId');
 		const datasetInputs = inputs.filter((input) => input.type === 'datasetId');
-
 		const modelInputs = inputs.filter((input) => input.type === 'modelId');
-		if (!modelInputs[0].value) {
+		const modelId = modelInputs?.[0]?.value?.[0];
+
+		if (modelId) {
+			let configs = await getModelConfigurationsForModel(modelId);
+			if (!configs[0].id) {
+				const model = await getModel(modelId);
+				if (model) await postAsConfiguredModel(model); // Create a model configuration if it does not exist
+				configs = await getModelConfigurationsForModel(modelId);
+			}
+			// Auto append output
+			if (configs[0].id) {
+				const config = configs[0];
+				state.transientModelConfig = config;
+				emit('update-state', state);
+				emit('append-output', {
+					type: ModelConfigOperation.outputs[0].type,
+					label: config.name,
+					value: config.id,
+					isSelected: false,
+					state: props.node.state
```

Comment:
This is not the same state object that you just modified (which is a cloned copy), I am guessing you want to emit the cloned version.